src/metadata.rs

```diff
@@ -146,12 +146,18 @@ fn read_flac(path: &Path) -> Result<SongTags> {
 		.get("DISCNUMBER")
 		.and_then(|d| d[0].parse::<u32>().ok());
 	let year = vorbis.get("DATE").and_then(|d| d[0].parse::<i32>().ok());
+	let streaminfo = tag.get_blocks(metaflac::BlockType::StreamInfo);
+	let duration = match streaminfo.first() {
+        Some(&&metaflac::Block::StreamInfo(ref s)) => (s.total_samples as u32 / s.sample_rate) as u32,
+        _ => 0
```

Comment:
I think the default match case should return None (duration unknown) rather than zero.

Reply:
nice! yeah totally missed this one